### PR TITLE
Added no_proxy autoconfiguration from rhsm conf and tests

### PR DESCRIPTION
**Pull request by ahitacat**
Signed-off-by: ahitacat <ahitacat@redhat.com>

### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [x] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### Complete Description of Additions/Changes:

Insights-client wasn't honoring the no_proxy configuration of `/etc/rhsm/rhsm.conf` file.

This PR adds the no_proxy value to the auto-configuration module, to retrieve the value of the rhsm configuration file. It provides some tests related to this.

This fix is related to: [rhbz#2065233]((https://bugzilla.redhat.com/show_bug.cgi?id=2065233)

### CODE REVIEW

File: insights/client/auto_config.py [hunk 1]
```diff
@@ -55,21 +55,25 @@ def verify_connectivity(config):
...
     saved_base_url = config.base_url
     if ca_cert is not None:
         saved_cert_verify = config.cert_verify
         config.cert_verify = ca_cert
     if proxy is not None:
         saved_proxy = config.proxy
         config.proxy = proxy
+    rhsm_no_proxy = None
+    if rhsm_no_proxy is not None or not '':
+        config.no_proxy = rhsm_no_proxy
```
Comment (strider) [inline insights/client/auto_config.py:76]: The default value of `rhsm_no_proxy` is `None` in the function parameters. So why resetting it to `None` again here? And it looks like we will always pass in the conditional here.

I guess the condition should be this one:

`if rhsm_no_proxy and rhsm_no_proxy != '':`

File: insights/client/auto_config.py [hunk 2]
```diff
@@ -128,6 +132,9 @@ def _try_satellite6_configuration(config):
         rhsm_proxy_port = rhsm_config.get('server', 'proxy_port').strip()
         rhsm_proxy_user = rhsm_config.get('server', 'proxy_user').strip()
         rhsm_proxy_pass = rhsm_config.get('server', 'proxy_password').strip()
+        rhsm_no_proxy = rhsm_config.get('server', 'no_proxy').strip()
+        if rhsm_no_proxy.lower() == 'None' or rhsm_no_proxy == '':
```
Comment (strider) [inline insights/client/auto_config.py:136]: `str.lower()` is supposed to return a copy of the string converted to lowercase and you compare it to `None`.

**Review by strider — CHANGES_REQUESTED** [review]

File: insights/client/auto_config.py (hunk 1 above)
Comment (ahitacat) [reply insights/client/auto_config.py:76]: you are right

File: insights/client/auto_config.py (hunk 2 above)
Comment (ahitacat) [reply insights/client/auto_config.py:136]: oh my mistake, it needs to be compare with `'none' `

File: insights/client/auto_config.py [hunk 3]
```diff
@@ -128,6 +131,9 @@ def _try_satellite6_configuration(config):
         rhsm_proxy_port = rhsm_config.get('server', 'proxy_port').strip()
         rhsm_proxy_user = rhsm_config.get('server', 'proxy_user').strip()
         rhsm_proxy_pass = rhsm_config.get('server', 'proxy_password').strip()
+        rhsm_no_proxy = rhsm_config.get('server', 'no_proxy').strip()
+        if rhsm_no_proxy.lower() == 'none' or rhsm_no_proxy == '':
```
Comment (strider) [inline insights/client/auto_config.py:135]: May be simplified as:

`if rhsm_no_proxy.lower() in ['none', '']:`

But it is just nitpicking here.

**Review by strider — APPROVED** [review]
Tested and it works, LGTM.

**Comment by ahitacat** [comment]
test me

**Pull request merged**